tools/install_mac_deps.sh

```diff
@@ -5,30 +5,38 @@
 set -e  # Exit on error.
 set -x  # Make command execution verbose
 
-# Install some custom requirements on OS X using brew
-BREW=/usr/local/bin/brew
-if [[ ! -x $BREW ]]; then
-    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
-fi
 
-BOOST_LIB="boost@1.76"
+# Set up Homebrew differently for arm64.
+if [[ $(uname -m) == 'arm64' ]]; then
+    (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> ~/.zprofile
+    eval "$(/opt/homebrew/bin/brew shellenv)"
+else
+    (echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >> ~/.zprofile
+    eval "$(/usr/local/bin/brew shellenv)"
+fi
 
-$BREW install autoconf automake bdw-gc ccache cmake \
-      libtool openssl pkg-config python coreutils
+if [[ ! -x brew ]]; then
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+fi
+HOMEBREW_PREFIX=$(brew --prefix)
 
-# We need to link boost.
-$BREW install ${BOOST_LIB}
-$BREW link ${BOOST_LIB}
-# Prefer Homebrew's bison and grep over the macOS-provided version
-$BREW install bison
-echo 'export PATH="/usr/local/opt/bison/bin:$PATH"' >> ~/.bash_profile
-$BREW install grep
-echo 'export PATH="/usr/local/opt/grep/libexec/gnubin:$PATH"' >> ~/.bash_profile
+# Install some custom requirements on OS X using brew
+BOOST_LIB="boost@1.84"
+brew install autoconf automake bdw-gc ccache cmake \
+      libtool openssl pkg-config coreutils bison grep \
```

Comment:
What's about flex? I remember macos shipped some ancient version by default, I do not recall the present status (esp. on builders)

Reply:
On Ventura laptop I'm having:
```
/usr/bin/flex --version
flex 2.6.4 Apple(flex-34)
```

This seems to be enough, ok.

Reply:
Huh, I have an older version on Sonoma (14.3.1):
```
flex 2.5.35 Apple(flex-32)
```

Reply:
Probably depends on installed xcode as well

Reply:
So, should I add a custom version? I believe in any case flex should be new enough.

Reply:
I think the default version is fine for recent versions of MacOS. I've built it locally on my system in the past with 2.5.35.

Reply:
Yes, default is fine now.